packages/trader/src/Modules/CFD/Containers/cfd-password-modal.tsx

```diff
@@ -27,18 +34,133 @@ import {
     WS,
 } from '@deriv/shared';
 import { localize, Localize } from '@deriv/translations';
-import SuccessDialog from 'App/Containers/Modals/success-dialog';
+import SuccessDialog from 'App/Containers/Modals/success-dialog.jsx';
 import 'Sass/app/modules/mt5/cfd.scss';
 import { connect } from 'Stores/connect';
 import ChangePasswordConfirmation from './cfd-change-password-confirmation';
 
+export type TCFDPasswordFormValues = { password: string };
+export type TCFDServerFormValues = { server: string };
+
+type TExtendedDetailsOfEachMT5Loginid = Omit<DetailsOfEachMT5Loginid, 'market_type'> & {
+    market_type?: 'synthetic' | 'financial' | 'gaming';
+};
+
+type TPasswordModalHeaderProps = {
+    should_set_trading_password: boolean;
+    should_show_server_form: boolean;
+    account_title: string;
+    is_password_reset_error: boolean;
+    platform: string;
+    has_mt5_account?: boolean;
+};
+
+type TIconTypeProps = {
+    platform: string;
+    type?: string;
+    is_eu: boolean;
+};
+
+type TCFDPasswordFormReusedProps = {
```

Comment:
I added this type to replace repeated properties that are reused in the hierarchy of 3 types which pass these properties down to the next one (TCFDPasswordFormProps -> TCFDCreatePasswordFormProps -> TCFDCreatePasswordProps) in order to avoid duplication.